packages/frontend/src/state/lp/hooks.ts

```diff
@@ -217,6 +217,164 @@ export const useCollectFees = () => {
+                                                                          .plus(sqrtSqueethPrice)
+                                                                          .minus(sqrtLowerPrice))
+        // Calculate amounts of each asset to LP
+        // x = L(sqrt(upperPrice) - sqrt(squeethPrice))) / sqrt(squeethPrice) * sqrt(upperPrice)
+        // y = L(sqrt(squeethPrice) - sqrt(lowerPrice))
+        wPowerPerpAmountInLPAfter = liquidity.times(sqrtUpperPrice.minus(sqrtSqueethPrice)).div((sqrtSqueethPrice.times(sqrtUpperPrice)))
+        wethAmountInLPAfter = liquidity.times(sqrtSqueethPrice.minus(sqrtLowerPrice))
+        const needMoreWeth = new BigNumber(wethAmountInLPBefore).lt(new BigNumber(wethAmountInLPAfter))
+        const needMoreSqueeth = new BigNumber(wPowerPerpAmountInLPBefore).lt(new BigNumber(wPowerPerpAmountInLPAfter))
+        tokenIn = needMoreWeth ? oSqueeth : weth
+        tokenOut = needMoreWeth ? weth : oSqueeth
+        amountIn = needMoreWeth && !needMoreSqueeth ? new BigNumber(wPowerPerpAmountInLPBefore).minus(new BigNumber(wPowerPerpAmountInLPAfter)).toFixed(0) :
+                   needMoreSqueeth && !needMoreWeth ? new BigNumber(wethAmountInLPBefore).minus(new BigNumber(wethAmountInLPAfter)).toFixed(0)
+                  : 0
+      }
+
+      // Get amount mins
+      const amount0New = isWethToken0 ? wethAmountInLPAfter : wPowerPerpAmountInLPAfter
+      const amount1New = isWethToken0 ? wPowerPerpAmountInLPAfter : wethAmountInLPAfter
+      const amount0MinNew = amount0New.times(new BigNumber(1).minus(slippage)).toFixed(0)
+      const amount1MinNew = amount1New.times(new BigNumber(1).minus(slippage)).toFixed(0)
+
+      // Get limit price
+      const amountOut = await getExactIn(new BigNumber(amountIn), tokenIn == oSqueeth)
+      const limitPrice = new BigNumber(amountOut).div(new BigNumber(amountIn)).times(new BigNumber(1).minus(slippage))
+
+      const abiCoder = new ethers.utils.AbiCoder()
+      const liquidatePreviousLP = {
+        // Liquidate LP
+        rebalanceLpInVaultType: new BigNumber(1).toFixed(0), // DecreaseLpLiquidity:
+        // DecreaseLpLiquidityParams: [tokenId, liquidity, liquidityPercentage, amount0Min, amount1Min]
+        data: abiCoder.encode(
+          ['uint256', 'uint256', 'uint256', 'uint128', 'uint128'],
+          [
+            uniTokenId,
+            position.liquidity,
+            fromTokenAmount(1, 18).toFixed(0),
+            amount0MinOld,
+            amount1MinOld,
+          ],
+        ),
+      }
+      const generalSwap = {
+        // Exchange necessary amount of oSQTH and ETH
+        rebalanceLpInVaultType: new BigNumber(5).toFixed(0), // GeneralSwap:
+        // GeneralSwap: [tokenIn, tokenOut, amountIn, limitPrice]
+        data: abiCoder.encode(
+          ['address', 'address', 'uint256', 'uint256', 'uint24'],
+          [tokenIn, tokenOut, amountIn, fromTokenAmount(limitPrice, 18).toFixed(0), POOL_FEE],
```

Comment:
pool fee right now can only be 0.3% based on how we use the quoter to figure out expected amts of swap